doc/sphinx/language/core/variants.rst

```diff
@@ -77,10 +82,12 @@ Definition by cases: match
    | @numeral
    | @string
 
-Objects of inductive types can be destructured by a case-analysis
-construction called *pattern matching* expression. A pattern matching
-expression is used to analyze the structure of an inductive object and
-to apply specific treatments accordingly.
+Note that the :n:`@pattern ::= @pattern10 : @term` production
+is not supported in :n:`match` patterns.  Trying to use it will give this error:
+
+.. exn:: in drop_notations_pattern: Casts are not supported in this pattern.
```

Comment:
Are we sure we want this error to be displayed as such?
I mean "in drop_notations_pattern:" is really for the dev of the code. Also, given the code is dropping something, couldn't it drop the cast as well? CC @herbelin.

Anyway, I believe "type cast" would be a more precise name for the thing that is not allowed.

Reply:
As a rule we show the exact full text of the error in the doc.  We could remove the `in drop_notations_pattern:` from the source code in this PR if you like.  Doing more than that is outside the scope of the PR, especially since I don't know the relevant code.

> Also, given the code is dropping something, couldn't it drop the cast as well

If we silently ignore `nat` in a `x : nat` pattern, users still need to know about that because they may try it and wonder why it doesn't work as they expect.  My impression from comments in the code is that that patterns could match on the type but that this was not implemented.

Improving the doc will make many of Coq's limitations more visible, perhaps leading to some of them being fixed.

Reply:
I think the pr should remove the function name from the error message.